pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/logicaltable/BaseLogicalTableIntegrationTest.java

```diff
@@ -0,0 +1,515 @@
+
+    String groovyQuery = "SELECT GROOVY('{\"returnType\":\"STRING\",\"isSingleValue\":true}', "
+        + "'arg0 + arg1', FlightNum, Origin) FROM mytable";
+
+    // Query should not throw exception
+    postQuery(groovyQuery);
+
+    // Disable groovy explicitly
+    queryConfig = new QueryConfig(null, true, null, null, null, null);
+
+    logicalTableConfig.setQueryConfig(queryConfig);
+    updateLogicalTableConfig(getLogicalTableName(), logicalTableConfig);
+
+    // grpc and http throw different exceptions. So only check error message.
+    Exception athrows = expectThrows(Exception.class, () -> postQuery(groovyQuery));
+    assertTrue(athrows.getMessage().contains("Groovy transform functions are disabled for queries"));
+
+    // Remove query config
+    logicalTableConfig.setQueryConfig(null);
+    updateLogicalTableConfig(getLogicalTableName(), logicalTableConfig);
+
+    athrows = expectThrows(Exception.class, () -> postQuery(groovyQuery));
+    assertTrue(athrows.getMessage().contains("Groovy transform functions are disabled for queries"));
+  }
+
+  @Test
+  public void testMaxQueryResponseSizeTableConfig()
+      throws Exception {
+    String starQuery = "SELECT * from mytable";
+
+    QueryConfig queryConfig = new QueryConfig(null, null, null, null, 100L, null);
+    LogicalTableConfig logicalTableConfig = getLogicalTableConfig(getLogicalTableName());
+    logicalTableConfig.setQueryConfig(queryConfig);
+    updateLogicalTableConfig(getLogicalTableName(), logicalTableConfig);
+
+    JsonNode response = postQuery(starQuery);
+    JsonNode exceptions = response.get("exceptions");
+    assertTrue(!exceptions.isEmpty()
+        && exceptions.get(0).get("errorCode").asInt() == QueryErrorCode.QUERY_CANCELLATION.getId());
+
+    // Query Succeeds with a high limit.
+    queryConfig = new QueryConfig(null, null, null, null, 1000000L, null);
+    logicalTableConfig.setQueryConfig(queryConfig);
+    updateLogicalTableConfig(getLogicalTableName(), logicalTableConfig);
+    response = postQuery(starQuery);
+    exceptions = response.get("exceptions");
+    assertTrue(exceptions.isEmpty(), "Query should not throw exception");
+
+    //Reset to null.
+    queryConfig = new QueryConfig(null, null, null, null, null, null);
+    logicalTableConfig.setQueryConfig(queryConfig);
+    updateLogicalTableConfig(getLogicalTableName(), logicalTableConfig);
+    response = postQuery(starQuery);
+    exceptions = response.get("exceptions");
+    assertTrue(exceptions.isEmpty(), "Query should not throw exception");
+  }
+
+  @Test
+  public void testMaxServerResponseSizeTableConfig()
+      throws Exception {
+    String starQuery = "SELECT * from mytable";
+
+    QueryConfig queryConfig = new QueryConfig(null, null, null, null, null, 1000L);
+    LogicalTableConfig logicalTableConfig = getLogicalTableConfig(getLogicalTableName());
+    logicalTableConfig.setQueryConfig(queryConfig);
+    updateLogicalTableConfig(getLogicalTableName(), logicalTableConfig);
```

Comment:
extract as helper method

Reply:
Only 2 lines of code is common across usage. Do you see a larger block of code being reused ?